Document pubternal API removal breaking change (#18406)
dotnet/docs: includes/core-changes/aspnetcore/3.0/pubternal-apis-removed.md

### "Pubternal" APIs removed

To better maintain the public API surface of ASP.NET Core, the majority of types in `*.Internal` namespaces (referred to as "pubternal" APIs) have become truly internal. Members in these namespaces were never meant to be supported as public-facing APIs. The APIs could break in minor releases and often did. Code that used to depend on these APIs breaks when updating to ASP.NET Core 3.0.

For more information, see [dotnet/aspnetcore#4932](https://github.com/dotnet/aspnetcore/issues/4932) and [dotnet/aspnetcore#11312](https://github.com/dotnet/aspnetcore/issues/11312).

#### Version introduced

3.0

#### Old behavior

> These APIs are marked with the `public` access modifier and exist in `*.Internal` namespaces.

Reviewer: ```suggestion
The affected APIs are marked with the `public` access modifier and exist in `*.Internal` namespaces.
```

#### New behavior

> These APIs are marked with the `internal` access modifier and can no longer be used.

Reviewer: ```suggestion
The affected APIs are marked with the [internal(~/docs/csharp/language-reference/keywords/internal.md) access modifier and can no longer be used by code outside that assembly.
```

"No longer be used by

#### Reason for change

The guidance for these "pubternal" APIs was that they:

* Could change without notice.

> * Weren't subject to our policies to prevent breaking changes.

Reviewer: ```suggestion
* Weren't subject to .NET policies to prevent breaking changes.
```

> It was determined that leaving the APIs `public` (even in the `*.Internal` namespaces) was still confusing to customers.

Reviewer: ```suggestion
Leaving the APIs `public` (even in the `*.Internal` namespaces) was confusing to customers.
```

#### Recommended action

Stop using these "pubternal" APIs. If you have questions about alternate APIs, open an issue in the [dotnet/aspnetcore](https://github.com/dotnet/aspnetcore/issues) repository.

For example, consider the following HTTP request buffering code in an ASP.NET Core 2.2 project. The `EnableRewind` extension method exists in the `Microsoft.AspNetCore.Http.Internal` namespace.

```csharp
HttpContext.Request.EnableRewind();
```

> In an ASP.NET Core 3.0 project, replace the `EnableRewind` with a call to the `EnableBuffering` extension method. The request buffering feature works as it did in the past. `EnableBuffering` calls the now `internal` API.

Reviewer: ```suggestion
In an ASP.NET Core 3.0 project, replace the `EnableRewind` call with a call to the `EnableBuffering` extension method. The request buffering feature works as it did in the past. `EnableBuffering` calls the now `internal` API.
```

```csharp
HttpContext.Request.EnableBuffering();
```

#### Category

ASP.NET Core

#### Affected APIs

> All APIs in the `Microsoft.AspNetCore.*` or `Microsoft.Extensions.*` namespaces that have an `Internal` segment in the namespace name. For example:

Reviewer: ```suggestion
All APIs in the `Microsoft.AspNetCore.*` and `Microsoft.Extensions.*` namespaces that have an `Internal` segment in the namespace name. For example:
```
